python/ingestion/dataset_utils.py

```diff
@@ -139,15 +142,41 @@ def merge_pop_numbers(df, demo, loc):
         'sex': std_col.SEX_COL,
     }
 
+    pop_dtype = {std_col.STATE_FIPS_COL: str,
+                 std_col.POPULATION_COL: object,
+                 std_col.POPULATION_PCT_COL: float}
+
     if demo not in on_col_map:
-        raise ValueError('%s not a demographic option, must be one of: %s' % (demo, list(on_col_map.keys())))
+        raise ValueError('%s not a demographic option, must be one of: %s' % (
+            demo, list(on_col_map.keys())))
 
     pop_table_name = 'by_%s_%s' % (demo, loc)
+
+    # all states, DC, PR
     if demo == 'race' and loc == 'state':
         pop_table_name += '_std'
 
-    pop_df = gcs_to_bq_util.load_df_from_bigquery('acs_population', pop_table_name, dtype={'state_fips': str})
-    pop_df = pop_df[[std_col.STATE_FIPS_COL, on_col_map[demo], std_col.POPULATION_COL, std_col.POPULATION_PCT_COL]]
+    pop_df = gcs_to_bq_util.load_df_from_bigquery(
+        'acs_population', pop_table_name, pop_dtype)
+    pop_df = pop_df[[std_col.STATE_FIPS_COL, on_col_map[demo],
+                     std_col.POPULATION_COL, std_col.POPULATION_PCT_COL]]
+
+    # other territories from ACS 2010 (VI, GU, AS, MP)
+    if loc == 'state':
+        verbose_demo = "race_and_ethnicity" if demo == 'race' else demo
+        pop_2010_table_name = 'by_%s_territory' % (
+            verbose_demo)
+
+        pop_2010_df = gcs_to_bq_util.load_df_from_bigquery(
+            'acs_2010_population', pop_2010_table_name, pop_dtype)
+
+        if len(pop_2010_df.index) > 0:
```

Comment:
Why is this check here? In what situation would we not want to merge it. It should always be there right?

Reply:
when it mocks the fetch call for ACS2010 in BRFSS it returns an empty df, so the column names were causing an error. Maybe would be better to generate fake territory data with the correct columns names for the UHC tetss, even though the territory data wont be merged?

Reply:
yeah, I would avoid adding more code to production just to get tests that are mocked differently than what we would expect to pass.

Reply:
That makes sense. I added the test mock for fetching ACS2010 data with just Guam so that it is more like the regular production flow.